### ISSUE TITLE
Unexpected gradient behavior for F.interpolate or upsample

### ISSUE THREAD

**Issue by jizongFox** [pytorch/pytorch#42772]
## To Reproduce

```
        features = torch.randn(1, 10, 256, 256, requires_grad=True) # a big grad-enabled feature map 
        reduced_feature = F.interpolate(features, size=(3, 3), mode="bilinear") # I reduced the size to (3,3)
        reduced_feature.retain_grad() 
        loss = reduced_feature[:, :, 1, 1].pow(2).sum() # I want to use the center of the reduced feature map to pass the gradient
        loss.backward()
        import matplotlib.pyplot as plt
        plt.subplot(121)
        plt.imshow(reduced_feature.grad[0, 0]) # the gradient is in the center, okay
        plt.title("gradient for the reduced feature space")
        plt.subplot(122)
        plt.imshow(features.grad[0].sum(0)) # the gradient should be in the large central part of the original feature map, as I have downsampled it. However, the gradient says that in fact, only one pixel in original feature affects. 
        plt.title("gradient for the original feature space")

        plt.show()
```
![image](https://user-images.githubusercontent.com/24821325/89702997-b0fa7a00-d914-11ea-9d84-9fb2e2d0c9b9.png)

## Expected behavior
![image](https://user-images.githubusercontent.com/24821325/89703024-fdde5080-d914-11ea-8fbf-ea9d11415904.png)
I think the gradient should be backpropagated to all the pixels which have been interpolated. (results realized by `adaptive_avg_pool2d`)

## Environment

PyTorch version: 1.5.1
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Ubuntu 20.04.1 LTS
GCC version: (Ubuntu 9.3.0-10ubuntu2) 9.3.0
CMake version: version 3.16.3

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: GeForce GTX 1050
Nvidia driver version: 450.57
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] numpydoc==0.9.2
[pip] pytorch-ranger==0.1.1
[pip] torch==1.5.1
[pip] torch-optimizer==0.0.1a13
[pip] torchvision==0.6.0a0+35d732a
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               10.1.243             h6bb024c_0  
[conda] mkl                       2020.0                      166  
[conda] mkl-service               2.3.0            py37he904b0f_0  
[conda] mkl_fft                   1.0.15           py37ha843d7b_0  
[conda] mkl_random                1.1.0            py37hd6b4f25_0  
[conda] numpy                     1.18.1           py37h4f9e942_0  
[conda] numpy-base                1.18.1           py37hde5b4d6_1  
[conda] numpydoc                  0.9.2                      py_0  
[conda] pytorch                   1.5.1           py3.7_cuda10.1.243_cudnn7.6.3_0    pytorch
[conda] pytorch-ranger            0.1.1                    pypi_0    pypi
[conda] torch-optimizer           0.0.1a13                 pypi_0    pypi
[conda] torchvision               0.6.1                py37_cu101    pytorch
## Additional context

cc @ezyang @SsnL @albanD @zou3519 @gqchen

**Comment by SsnL** [comment]
isn't this expected? you are doing bilinear downsampling, which means that each output pixel is only determined by at most 4 input pixels.

**Comment by jizongFox** [comment]
> isn't this expected? you are doing bilinear downsampling, which means that each output pixel is only determined by at most 4 input pixels.
Good point. I think it depends on the interpretations. As here I strongly downsampled the feature map and the downsampling method may be supposed to perform information aggregation. Yeh. you may be right in the case of bilinear downsampling. What if I want a downsampling result to represent more points in the original feature space instead of a single point?
